60_network_learning.Rmd

```diff
@@ -0,0 +1,654 @@
+
+6.  **Transformation into similarity / edge weight:** Finally, the dissimilarities are transformed into similarities by $s_{ij} = 1 - d_{ij}$, which are used as edge weights. Thus, the similarity matrix is equal to the adjacency matrix in a weighted network.
+
+In the following, we show an application for each of the three types of association: correlation, proportionality, and partial correlation. 
+
+We demonstrate the workflow using the the **PeerJ data set** [@potbhare2022skin]. It contains skin microbial profiles of 58 subjects.
+
+```{r, message=FALSE, warning=FALSE}
+library(mia)
+```
+
+```{r load_data}
+data("peerj13075", package = "mia")
+tse <- peerj13075
+dim(tse)
+```
+
+```{r, echo=FALSE}
+# The networks shown in this chapter take several minutes to generate. 
+# Therefore, the network objects have been stored in a folder and are loaded 
+# here to reduce the time needed to knit the book.
+load("general/network_data/networks.RData")
+```
+
+### Install packages
+
+Three packages used in this section are available on GitHub only: `SpiecEasi`, `SPRING`, and `NetCoMi`. We recommend that you install these packages before proceeding.
+
+```{r install_packages}
+if(!require(SpiecEasi)){
+  devtools::install_github("zdk123/SpiecEasi")
+}
+
+if(!require(SPRING)){
+  devtools::install_github("GraceYoon/SPRING")
+}
+
+if(!require(NetCoMi)){
+  devtools::install_github("stefpeschel/NetCoMi", force = TRUE,
+                           dependencies = c("Depends", "Imports", "LinkingTo"),
+                           repos = c("https://cloud.r-project.org/",
+                                     BiocManager::repositories()))
+}
+```
+
+
+### Data preparation
+
+Before we apply the network learning methods, we perform some data preparation steps similar to that in section \@ref(differential-abundance):
+
+-   Aggregate to genus level (would not be necessary here since the data is already at genus level)
+-   Prevalence filtering (keep genera with prevalence \> 20%)
+-   Add assay with clr-transformed abundances
+
+```{r preprocessing}
+# Agglomerate by genus and subset by prevalence
+tse <- subsetByPrevalentFeatures(tse,
+                                 rank = "genus",
+                                 prevalence = 0.2,
+                                 detection = 0)
+
+# Add clr-transformed abundances
+tse <- transformAssay(tse, method="clr", pseudocount=1)
+
+dim(tse)
+```
+
+Since we want to use the phylum names for node colors later in the network plot, we assign low abundant phyla to a new class "Other" to reduce the number of colors needed in the plot.
+
+```{r}
+# Assign low abundant phyla to a "Other" class
+phyla <- rowData(tse)$phylum
+low_abundant <- names(table(phyla)[table(phyla) <= 3])
+phyla[phyla %in% low_abundant] <- "Other"
+rowData(tse)$phylum <- phyla
+
+# Take a look at the table
+table(rowData(tse)$phylum)
```

Comment:
I suggest two small changes for clarity. Ok?

1. Now "rare" phyla are detected based on how many *rows* they have; this counts the number of different ASVs (or other features) within the phylum, and not their abundance although the text - in my view misleadingly - says that these would be "low abundant phyla"; this is not necessarily the case. I think that abundance/prevalence-based agglomeration would be better justified? For that you could do 

```
# Detect rare features (this is complement of prevalent)
rare.taxa <- getRareFeatures(tse, rank="Phylum", detection=0, prevalence=20/100)

# Merge non-prevalent features into "Other" group
rowData(tse)$Phylum_merged <- stringi::stri_replace_all_regex(rowData(tse)$Phylum, pattern=rare.taxa, replacement="Other", vectorize=FALSE)

# There may also be NA phyla - decide what to do with those; perhaps just discard those rows where phylum is not identified?
```



2. Adding a new column, for instance phylum_merged, instead of overwriting original phyla. Then both will remain available.